### PR TITLE
CONTRIBUTING changes: Command line & conflict resolution

### PR DESCRIPTION
**Pull request by quill18**
In an effort to help combat some of the common PR issues we keep seeing, I have produced a new video-tutorial and updated CONTRIBUTING.md to reflect the example workflow.

This explains how to pull changes from upstream, how to rebase, and how to resolve merge conflicts.

### CODE REVIEW

**Comment by vogonistic** [comment]
@quill18 In case you forgot it, the youtube video is still private.

**Comment by quill18** [comment]
@vogonistic It's actually still processing -- also, I wanted to get feedback on the workflow before merging this request.  

The video has been set to unlisted (though you may still have to wait for YouTube to process it):   https://www.youtube.com/watch?v=R2fl17eEpwI

**Comment by bjubes** [comment]
lol i guess we had the same idea. I added a [rebase tutorial on the wiki](https://github.com/TeamPorcupine/ProjectPorcupine/wiki/Git-Problems-and-Fixes)

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -4,24 +4,63 @@ If you would like to contribute to this project by modifying/adding to the progr
...
+Edit the conflicting file(s) and look for a block that looks like this:  
+    `<<<<<<< HEAD`  
+    `my awesome change`  
+    `=======`  
+    `some other person's less awesome change`  
+    `>>>>>>> some-branch`  
+
+Replace all five (or more) lines with the correct version (yours, theirs, or
+a combination of the two).  ONLY the correct content should remain (none of 
+that "<<<<< HEAD" stuff.)
```
Comment (vogonistic) [inline CONTRIBUTING.md]: Maybe add something here showing the potential outcome, like:

Example after merge:
```
    `some other person's less awesome change`
    `my awesome change`
```

**Comment by vogonistic** [comment]
I'll review the youtube video when it's finished processing, but the merge instructions look good. I ran through the commands to check.

**Comment by Mizipzor** [comment]
Great initiative! I was hoping for more of this. I mentioned it in other threads, if you're in this project you shouldn't just learn Unity and C#. Learning a version control system is today equally essential to a game developer.

**Comment by Dormanil** [comment]
You're explaining twice how to pull the latest and greatest bits from upstream/master, but as far as I can tell until now, everything else is fine in the video. Continue with it now.

**Comment by bjubes** [comment]
the video is only at 360p quality. I assume this is because you wanted to quickly upload it to show it to us first, but if not make sure to upload the HD version to the channel

**Comment by TheRiven** [comment]
@bjubes the quality will improve you just have to give it time to finish

**Comment by vogonistic** [comment]
Moving the comments I have on the video to the correct issue =)

@quill18 I think you could probably spend a couple of hours talking about what the git tree looks like so that people understand the difference between `upstream`, `origin` and the 3 potential versions of `master`. It's hard to capture it in just 1h, but you did a pretty good job going through everything. People who have trouble follow it will just have to ask questions.

The biggest omission in the video is that you don't show people how to view what changes will be included in the PR before it is created. On the page where you set the title and body of the PR, below the body text field you can see the commits and the changes included. If more people reviewed that before creating the PR, we'd have less of them with problems.

**Comment by quill18** [comment]
@vogonistic Arg!  You're absolutely right.  I take it for granted to check the commits and -- more importantly -- files before submitting.  I'll see if I can slip in an edit tomorrow.

**Comment by quill18** [comment]
I forgot I hadn't merged this yet and deleted the branch too soon.  I want to leave this open until tomorrow for any additional feedback before I update & merge.

**Comment by Mizipzor** [comment]
@quill18 May I suggest annotations in the video or links in the description that jumps you straight to a certain topic? Git is difficult and I would imagine that some people want to re-watch some segment virtually every time they do it. 

Also, sometimes images are helpful and I think [this page](https://www.atlassian.com/git/tutorials/merging-vs-rebasing/summary) does and excellent job of explaining it.

**Pull request merged**